lib/minitest/queue.rb

```diff
@@ -4,6 +4,46 @@
 require 'minitest/reporters'
 
 module Minitest
+  class Requeue < Skip
+    attr_reader :failure
+
+    def initialize(failure)
+      super()
+      @failure = failure
+    end
+
+    def result_label
+      "Requeued"
+    end
+
+    def backtrace
+      failure.backtrace
+    end
+
+    def error
+      failure.error
+    end
+
+    def message
+      failure.message
+    end
+  end
+
+  module Requeueing
+    # Make requeues acts as skips for reporters not aware of the difference.
+    def skipped?
+      super || requeued?
+    end
```

Comment:
It's because of how the `skipped?` method is implemented in Minitest: https://github.com/seattlerb/minitest/blob/174a860e6c18162ebeb52c198dd45eb55eb8afcb/lib/minitest/test.rb#L246-L248

That code base is incredibly hard to extend.